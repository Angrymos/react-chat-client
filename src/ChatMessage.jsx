import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

const styles = theme => ({
    messageWrapper: {
        display: 'flex',
        justifyContent: 'flex-start',
        alignItems: 'center',
        padding: `${theme.spacing.unit}px ${theme.spacing.unit * 3}px`,
    },
    messageWrappperFromMe: {
        justifyContent: 'flex-end',
    },
    message: {
        maxWidth: '70%',
        minWidth: '10%',
        padding: theme.spacing.unit,
        marginLeft: theme.spacing.unit * 2,
    },
    messageFromMe: {
        marginRight: theme.spacing.unit * 2,
        backgroundColor: '#e6dcff',
    },
    statusMessage: {
        width: '100%',
        textAlign: 'center',
    },
    statusMessageUser: {
        display: 'inline',
    },
});

function PaperSheet(props) {
    const { classes } = props;

    return (
        <div>
            <Paper className={classes.messageFromMe}>
                <Typography variant="caption">
                    {'ann'}
                </Typography>
                <Typography variant="body1">{'Some message'}</Typography>
                <Typography variant="caption" className={''}>
                    1 min ago.
                </Typography>
            </Paper>
        </div>
    );
}

export default withStyles(styles)(PaperSheet);
