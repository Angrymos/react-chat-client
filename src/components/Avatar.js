import React from 'react';
import MUIAvatar from '@material-ui/core/Avatar';
import getColor from '../utils/color-from';
import titleInitials from '../utils/title-initial';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  row: {
    display: 'flex',
    justifyContent: 'center',
  },
};

const Avatar = ({ classes, colorFrom, children, ...rest }) => (
  <div className={classes.row}>
    <MUIAvatar style={{ backgroundColor: getColor(colorFrom), margin: 10 }} {...rest}>
      {titleInitials(children)}
    </MUIAvatar>
  </div>
);

export default withStyles(styles)(Avatar);
