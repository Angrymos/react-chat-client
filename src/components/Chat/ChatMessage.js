import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import Avatar from '@material-ui/core/Avatar';
import classnames from 'classnames';
import LetterAvatar from '../LetterAvatar';

const styles = theme => ({
  messageWrapper: {
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    padding: `${theme.spacing.unit}px ${theme.spacing.unit * 3}px`,
  },
  messageWrappperFromMe: {
    justifyContent: 'flex-end',
  },
  message: {
    maxWidth: '45%',
    minWidth: '10%',
    padding: theme.spacing.unit,
  },
  messageFromMe: {
    marginRight: theme.spacing.unit * 2,
    backgroundColor: '#e6dcff',
  },
});

const ChatMessage = ({ classes, message }) => {
  const userAvatar = (
    <LetterAvatar
      sender={message.sender}
    />
  );

  const isMessageFromMe = message.sender === 'me';

  return (
    <div
      className={classnames(
        classes.messageWrapper,
        isMessageFromMe && classes.messageWrappperFromMe,
      )}
    >
      {isMessageFromMe && userAvatar}
      <Paper
        className={classnames(
          classes.message,
          isMessageFromMe && classes.messageFromMe,
        )}
      >
        <Typography variant='caption'>
          {message.sender}
        </Typography>
        <Typography variant='body1'>
          {message.content}
        </Typography>
        <Typography variant='caption'>
          1 min ago.
        </Typography>
      </Paper>
      {!isMessageFromMe && userAvatar}
    </div>
  );
};

export default withStyles(styles)(ChatMessage);
