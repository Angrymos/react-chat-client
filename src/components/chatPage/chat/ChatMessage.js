import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import classnames from 'classnames';
import Avatar from '../../Avatar';

const styles = theme => ({
  messageWrapper: {
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    padding: `${theme.spacing.unit}px ${theme.spacing.unit * 3}px`,
  },
  messageWrappperFromMe: {
    justifyContent: 'flex-end',
  },
  message: {
    maxWidth: '45%',
    minWidth: '10%',
    padding: theme.spacing.unit,
  },
  messageFromMe: {
    marginRight: theme.spacing.unit * 2,
    backgroundColor: '#e6dcff',
  },
});

const ChatMessage = ({ classes, message }) => {
  const userAvatar = (
    <Avatar colorFrom={message.sender}>{message.sender}</Avatar>
  );

  const isMessageFromMe = message.sender === 'me';

  return (
    <div
      className={classnames(
        classes.messageWrapper,
        isMessageFromMe && classes.messageWrappperFromMe,
      )}
    >
      {isMessageFromMe && userAvatar}
      <Paper
        className={classnames(
          classes.message,
          isMessageFromMe && classes.messageFromMe,
        )}
      >
        <Typography variant='caption'>
          {message.sender}
        </Typography>
        <Typography variant='body1'>
          {message.content}
        </Typography>
        <Typography variant='caption'>
          1 min ago.
        </Typography>
      </Paper>
      {!isMessageFromMe && userAvatar}
    </div>
  );
};

export default withStyles(styles)(ChatMessage);
