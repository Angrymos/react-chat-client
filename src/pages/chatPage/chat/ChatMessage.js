import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import classnames from 'classnames';
import Avatar from '../../../components/Avatar';
import randomColor from '../../../utils/color-from';
import moment from 'moment';

const styles = theme => ({
  messageWrapper: {
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    padding: `${theme.spacing.unit}px ${theme.spacing.unit * 3}px`,
  },
  messageWrappperFromMe: {
    justifyContent: 'flex-end',
  },
  message: {
    maxWidth: '45%',
    minWidth: '10%',
    padding: theme.spacing.unit,
  },
  messageFromMe: {
    marginRight: theme.spacing.unit * 2,
    backgroundColor: '#e6dcff',
  },
  statusMessage: {
    width: '100%',
    textAlign: 'center',
  },
  statusMessageUser: {
    display: 'inline',
  },
});

const ChatMessage = ({ classes, message, activeUserId }) => {
  const userAvatar = (
    <Avatar colorFrom={message.sender.username}>{message.sender.username}</Avatar>
  );

  if (message.statusMessage) {
    return (
      <div className={classes.messageWrapper}>
        <Typography className={classes.statusMessage}>
          <Typography
            variant='caption'
            style={{ color: randomColor(message.sender._id) }}
            className={classes.statusMessageUser}
          >
            {message.sender.username}
          </Typography>
          {message.content}
          <Typography variant='caption' component='span'>
            {moment(message.createdAt).fromNow()}
          </Typography>
        </Typography>
      </div>
    );
  }

  const isMessageFromMe = message.sender._id === activeUserId;

  return (
    <div
      className={classnames(
        classes.messageWrapper,
        isMessageFromMe && classes.messageWrappperFromMe,
      )}
    >
      {isMessageFromMe && userAvatar}
      <Paper
        className={classnames(
          classes.message,
          isMessageFromMe && classes.messageFromMe,
        )}
      >
        <Typography variant='caption'>
          {message.sender.username}
        </Typography>
        <Typography variant='body1'>
          {message.content}
        </Typography>
        <Typography variant='caption'>
          1 min ago.
        </Typography>
      </Paper>
      {!isMessageFromMe && userAvatar}
    </div>
  );
};

ChatMessage.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,

  message: PropTypes.shape({
    chatId: PropTypes.string.isRequired,
    content: PropTypes.string.isRequired,
    sender: PropTypes.object.isRequired,
    createdAt: PropTypes.string.isRequired,
    statusMessage: PropTypes.bool
  }),
  activeUserId: PropTypes.string.isRequired,
};

export default withStyles(styles)(ChatMessage);
