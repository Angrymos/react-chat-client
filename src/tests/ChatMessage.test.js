import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatMessage from '../pages/chatPage/chat/ChatMessage';
import Avatar from '../components/Avatar';

const mockProps = {
  message: {
    chatId: '5bd82f4bac450f23acdd78ad',
    content: 'Hello',
    sender: {
      id: '5bd6dda2578d3404b06c58fb',
      username: '123',
      lastName: '',
      firstName: '',
    },
    statusMessage: false,
    createdAt: '2018-03-16T10:53:23.200Z',
  },

  activeUserId: '5bd6dda2578d3404b06c58fb',
};

jest.mock('moment', () => () => ({ fromNow: () => '2 days ago' }));

describe('<Avatar />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatMessage {...mockProps} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders name of other users correctly', () => {
    const tree = renderer
      .create(<ChatMessage
        {...mockProps}
        sender={{
          _id: '43egfd24',
          username: 'someone',
          firstName: '',
          lastName: '',
        }}
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders status message', () => {
    const message = {
      ...mockProps.message,
      content: ' joined',
      statusMessage: true,
    };
    
    const props = {
      ...mockProps,
      message: message,
    };

    const tree = renderer.create(<ChatMessage {...props} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});



