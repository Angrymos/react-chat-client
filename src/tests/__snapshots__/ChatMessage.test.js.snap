// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> renders name of other users correctly 1`] = `
<div
  className="ChatMessage-messageWrapper-64"
>
  <div
    className="MuiPaper-root-70 MuiPaper-elevation2-74 MuiPaper-rounded-71 ChatMessage-message-66"
  >
    <span
      className="MuiTypography-root-97 MuiTypography-caption-107"
    >
      123
    </span>
    <p
      className="MuiTypography-root-97 MuiTypography-body1-106"
    >
      Hello
    </p>
    <span
      className="MuiTypography-root-97 MuiTypography-caption-107"
    >
      1 min ago.
    </span>
  </div>
  <div
    className="Avatar-row-123"
  >
    <div
      className="MuiAvatar-root-124 MuiAvatar-colorDefault-125"
      style={
        Object {
          "backgroundColor": "#f44336",
          "margin": 10,
        }
      }
    >
      1
    </div>
  </div>
</div>
`;

exports[`<Avatar /> renders status message 1`] = `
<div
  className="ChatMessage-messageWrapper-64"
>
  <p
    className="MuiTypography-root-97 MuiTypography-body1-106 ChatMessage-statusMessage-68"
  >
    <span
      className="MuiTypography-root-97 MuiTypography-caption-107 ChatMessage-statusMessageUser-69"
      style={
        Object {
          "color": "#607d8b",
        }
      }
    >
      123
    </span>
     joined
    <span
      className="MuiTypography-root-97 MuiTypography-caption-107"
    >
      2 days ago
    </span>
  </p>
</div>
`;
